src/oatpp/network/tcp/server/ConnectionProvider.cpp

```diff
@@ -43,6 +43,29 @@
   #endif
 #endif
 
+#if _WIN32_WINNT < 0x0600
```

Comment:
Could you wrap the whole `#if` block in another check for an MinGW build like `#ifdef __MINGW32__
`? For linux compiler `#if _WIN32_WINNT < 0x0600` evaluates to true because the `_WIN32_WINNT` macro is not set.